scio-core/src/main/scala/com/spotify/scio/io/Tap.scala

```diff
@@ -88,9 +87,18 @@ private[scio] class MaterializeTap[T: Coder] private (val path: String, coder: B
   private val _path = ScioUtil.addPartSuffix(path)
 
   override def value: Iterator[T] = {
-    FileStorage(_path)
-      .avroFile[GenericRecord](AvroBytesUtil.schema)
-      .map(AvroBytesUtil.decode(coder, _))
+    val storage = FileStorage(_path)
+
+    if (storage.isDone) {
+      storage
+        .avroFile[GenericRecord](AvroBytesUtil.schema)
+        .map(AvroBytesUtil.decode(coder, _))
+    } else {
+      throw new RuntimeException(
```

Comment:
@ClaireMcGinty when will we hit this case? users can't start reading taps if the pipeline execution is not done.

Reply:
yeah the regular ScioIO taps are very safe! But for example, this compiles and might not look obviously wrong to the user if they don't know how `materialize` works under the hood:

```
val data = sc.parallelize(1 to 10)
sc.run().waitUntilDone().tap(data.materialize)
```

Reply:
Ahh yes I think they could (somehow I missed the example in the comment) ... can we just reword this and not mention the closing?

Reply:
ah yes my bad! Meant to reword that before merging this 😅I'm going to make a PR tomorrow rewording a bunch of the docs/comments that mention closing, I'll include this :)